automation/devops_automation_infra/k8s_plugins/kafka.py

```diff
@@ -30,7 +53,18 @@ def _expose(self):
             advertised_brokers['brokers'].append({'broker': i, 'advertisedHost': self._master.ip})
 
         kafka_spec['kafka']['listeners']['external'] = {'type': 'nodeport', 'tls': False, 'overrides': advertised_brokers}
-        self._is_exposed = True
+
+        pods_timestamps = [pod.status.start_time for pod in self._list_broker_pods()]
+        custom_object_client.patch_namespaced_custom_object(namespace=self._namespace,
+                                                            group='kafka.strimzi.io',
+                                                            version='v1beta1',
+                                                            plural='kafkas',
+                                                            name=self._name,
+                                                            body={'spec': kafka_spec})
+
+        logging.debug("Waiting for kafka brokers to restart")
+        waiter.wait_for_predicate(lambda: [pod.status.start_time for pod in self._list_broker_pods()] != pods_timestamps, timeout=30)
```

Comment:
there is an issue with this method . what happens if one broker restatred 
i think this should be extracted to 
```
brokers_state = self._brokers_state()
HERE the restart code .. 

``` python
waiter.wait_for_predicate(lambda: _kafka_brokers_restarted(brokers_state)

def _brokers_state():
    {pod.metadata.name : pod.status.start_time}

def _kafka_brokers_restarted(old_brokers_state):
    state=self._brokers_state()
    return len(old_brokers_state) == len(state) and len(a1.items() & b1.items()) == 0
```